packages/protocol/contracts/stability/SortedOracles.sol

```diff
@@ -308,6 +308,16 @@ contract SortedOracles is ISortedOracles, ICeloVersionedContract, Ownable, Initi
     return tokenReportExpirySeconds[token];
   }
 
+  /**
+   * @notice Returns the identifier (address) from a human readable currency pair (eg. "CELO/BTC")
+   *         Can be used from other contracts to compute the identifier to read rates.
+   * @param currencyPair The string human readable currency pair
+   * @return The identifier as an address
+   */
+  function getCurrencyPairIdentifier(string calldata currencyPair) external pure returns (address) {
+    return address(bytes20(keccak256(abi.encodePacked(currencyPair))));
```

Comment:
note that this uses the left 20 bytes (bytes 0-20), while address convention uses the right 20 bytes (bytes 12-32)

Reply:
to use the right 20 bytes you need to do `address(uint160(uint256(bytes32())))`

Reply:
this is probably not important for your use case, so more of a nit than a bug

Reply:
Yeah this was intentional to match with the implementation we did off-chain. So If I understand correctly we could have done: `address(keccak256(abi.encodePacked(currencyPair)))` but that would result in bytes 12-32 right?

Reply:
that'll give you a `TypeError: Explicit type conversion not allowed from "bytes32" to "address".`

Solidity allows EITHER a type cast (between `bytesX`, `uintX`, `address`, etc) OR a length cast (`uintX` to `uintY` or `bytesX` to `bytesY`) for each explicit cast. To get the right aligned bytes we need to cast through uint (uints are right-aligned while bytes are left-aligned) and that gives us  this annoying multi-step cast `bytes32` => `uint256` => `uint160` => `address`.

Reply:
Ah ok now I saw your 2nd and 3rd message (didn't see them in the UI when I initially answered) and also re-parsed your 1st (read `address conversion` instead of `address convention`, thus my `address(keccak...)`).

Ok so I'm hearing that there's a convention that addresses should be extracted as the last 20 bytes of a byte array in any situation where we're turning an arbitrary (>20 bytes) array in an address, and that it would be best to apply this here as well. Is that correct?

Reply:
Yes. All address types (external account, create, and create2) are computed as the final 12 bytes of a keccak256 digest. It would be good to follow that convention if using the address type

Reply:
Cool, somehow I didn't think of it in terms of creating an address because it's just an abstract identifier but it makes sense to keep it consistent. I made the changes to use the last 20bytes 🚀